### PR TITLE
(🎁) Support only vararg in custom converters

### PR DESCRIPTION
**Pull request by KotlinIsland**
This allows registering a custom converter that only has vararg arguments

### CODE REVIEW

File: atest/testdata/keywords/type_conversion/custom_converters.robot [hunk 1]
```diff
@@ -69,6 +69,9 @@ Failing conversion
     Conversion should fail    Strict    wrong type
     ...    type=Strict    error=TypeError: Only Strict instances are accepted, got string.
 
+Only vararg
+    Only var arg    10    ${10}
```
Comment (pekkaklarck) [inline atest/testdata/keywords/type_conversion/custom_converters.robot:73]: This test needs a counterpart under `atest/robot`. See `atest/README.rst` for details.

File: src/robot/running/arguments/argumentspec.py [hunk 2]
```diff
@@ -55,6 +55,10 @@ def minargs(self):
     def maxargs(self):
         return len(self.positional) if not self.var_positional else sys.maxsize
 
+    @property
+    def first_positional(self):
+        return self.positional and self.positional[0] or self.var_positional or None
```
Comment (pekkaklarck) [inline src/robot/running/arguments/argumentspec.py:60]: This property is somewhat confusingly named and doesn't feel too useful in general. Better to remove it and inline this code to the only place where it's used.

**Review by pekkaklarck — CHANGES_REQUESTED** [review]
Separate test is needed to validate that if a converter accepts varargs, the second argument is the library itself. For other changes see the inline comments.

File: atest/testdata/keywords/type_conversion/custom_converters.robot (hunk 1 above)
Comment (KotlinIsland) [reply atest/testdata/keywords/type_conversion/custom_converters.robot:73]: Thanks, done!

File: src/robot/running/arguments/argumentspec.py (hunk 2 above)
Comment (KotlinIsland) [reply src/robot/running/arguments/argumentspec.py:60]: Thanks, done!

**Pull request closed without merging**